src/gui/plugins/visualize_frustum/VisualizeFrustum.cc

```diff
@@ -0,0 +1,447 @@
+/*
+ * Copyright (C) 2025 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include "VisualizeFrustum.hh"
+
+#include <cstddef>
+#include <mutex>
+#include <ostream>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include <sdf/Link.hh>
+#include <sdf/Model.hh>
+
+#include <gz/common/Console.hh>
+#include <gz/common/Profiler.hh>
+
+#include <gz/plugin/Register.hh>
+
+#include <gz/math/Vector3.hh>
+#include <gz/math/Pose3.hh>
+
+#include <gz/transport/Node.hh>
+
+#include <gz/gui/Application.hh>
+#include <gz/gui/Conversions.hh>
+#include <gz/gui/GuiEvents.hh>
+#include <gz/gui/MainWindow.hh>
+
+#include "gz/sim/components/Name.hh"
+#include "gz/sim/components/World.hh"
+#include "gz/sim/EntityComponentManager.hh"
+#include "gz/sim/Entity.hh"
+#include "gz/sim/rendering/RenderUtil.hh"
+
+#include "gz/rendering/RenderTypes.hh"
+#include "gz/rendering/RenderingIface.hh"
+#include "gz/rendering/RenderEngine.hh"
+#include "gz/rendering/Scene.hh"
+#include "gz/rendering/FrustumVisual.hh"
+
+#include "gz/sim/components/Link.hh"
+#include "gz/sim/components/Sensor.hh"
+#include "gz/sim/components/Model.hh"
+#include "gz/sim/components/ParentEntity.hh"
+#include "gz/sim/components/Pose.hh"
+#include "gz/sim/Util.hh"
+
+namespace gz
+{
+namespace sim
+{
+inline namespace GZ_SIM_VERSION_NAMESPACE
+{
+  /// \brief Private data class for VisualizeFrustum
+  class VisualizeFrustumPrivate
+  {
+    /// \brief Transport node
+    public: transport::Node node;
+
+    /// \brief Scene Pointer
+    public: rendering::ScenePtr scene;
+
+    /// \brief Pointer to FrustumVisual
+    public: rendering::FrustumVisualPtr frustum;
+
+    /// \brief URI sequence to the frustum link
+    public: std::string frustumString{""};
+
+    /// \brief LaserScan message from sensor
+    public: msgs::LogicalCameraSensor msg;
+
+    /// \brief Pose of the frustum visual
+    public: math::Pose3d frustumPose{math::Pose3d::Zero};
+
+    /// \brief Topic name to subscribe
+    public: std::string topicName{""};
+
+    /// \brief List of topics publishing LogicalCameraSensor messages.
+    public: QStringList topicList;
+
+    /// \brief Entity representing the sensor in the world
+    public: sim::Entity frustumEntity;
+
+    /// \brief Mutex for variable mutated by the checkbox 
```

Comment:
CI picked up a minor trailing white space issue here:

```suggestion
    /// \brief Mutex for variable mutated by the checkbox
```